pyvista/plotting/renderer.py

```diff
@@ -874,7 +874,7 @@ def add_actor(
         self.AddActor(actor)  # must add actor before resetting camera
         self._actors[name] = actor
 
-        if reset_camera or not self.camera_set and reset_camera is None and not rv:
+        if reset_camera or (not self.camera_set and reset_camera is None and not rv):
```

Comment:
This needs careful review, I don't think the assumption here by ruff is correct

Reply:
While ruff has not changed the python logic, it highlights the underlying issue.  The correct logic here is dense and not immediately obvious to me.  Could be a bug here....

Reply:
`reset_camera` is declared to be of type `bool`, so I'm not sure when it can be `None`.

Reply:
We now have redundant logic and not reachable checks enabled with mypy, merge with main might highlight this?

Reply:
If we change `reset_camera=False` to `reset_camera: bool=False` in the function signature, mypy indeed warns that `reset_camera is None` is unreachable. We can remove this logic altogether is it must be `False` here.

And `rv` is initialized earlier as `None` and will only be set if an actor of the same name previously existed and was removed. So this logic will reset the camera if it hasn't already been set and if no actors were removed.

I think this is pretty much what the logic was already doing before this PR:
```suggestion
        if reset_camera or (not self.camera_set and rv is None):
```

But, is this correct...? We only want to reset the camera if explicitly requested OR if it hasn't been set and no actor was removed? Shouldn't we reset the camera if the camera _is_ set and an actor _is_ removed?

I'm thinking that this may be the correct logic here, since we probably need to reset the camera if an actor was removed:
```suggestion
        if reset_camera or (self.camera_set and rv is not None):
```